pkg/rehearse/jobs.go

```diff
@@ -714,7 +714,32 @@ func SelectJobsForChangedRegistry(regSteps []registry.Node, allPresubmits presub
 }
 
 func moreRelevant(one, two *config.DataWithInfo) bool {
-	return one.Info.Filename > two.Info.Filename
+	branchI := one.Info.Metadata.Branch
+	branchJ := two.Info.Metadata.Branch
+	if branchI == "master" || branchI == "main" {
+		return true
+	} else if strings.HasPrefix(branchI, "release-4.") || strings.HasPrefix(branchI, "openshift-4.") {
+		if strings.HasPrefix(branchJ, "release-4.") || strings.HasPrefix(branchJ, "openshift-4.") {
+			numberI := strings.TrimPrefix(branchI, "release-4.")
+			if numberI == branchI {
+				numberI = strings.TrimPrefix(branchI, "openshift-4.")
+			}
+			numberJ := strings.TrimPrefix(branchJ, "release-4.")
+			if numberJ == branchJ {
+				numberJ = strings.TrimPrefix(branchJ, "openshift-4.")
+			}
+			intI, _ := strconv.Atoi(numberI)
+			intJ, _ := strconv.Atoi(numberJ)
+			if intI > intJ {
+				return true
+			} else {
+				return false
+			}
+		} else {
+			return true
+		}
+	}
+	return branchI > branchJ
```

Comment:
I was thinking about this, and I think we'll want to optimize this - because it will be used in a less function of a sort algorithm, it will potentially be called many times, so we can try being a bit smart about this. How about pregenerating  a map from a known branch names to numbers? You would do something like this once:

```go
branches := map[string]int{}
for i := 1; i < 50; i++ {
  branches[fmt.Sprintf("release-4.%d", i)] = i
  branches[fmt.Sprintf("openshift-4.%d", i)] = i
}
```

And then you'd only used that map instead of parsing the strings all the time. You could go even further, add `main` and `master` mapped to `100` or something. And you could even more further, consider the "not in map" case (which will return the default value of `0`) too. Then you could reduce the whole `moreRelevant` to a comparison of integers fetched from a map.

Reply:
> Then you could reduce the whole moreRelevant to a comparison of integers fetched from a map.

I don't know how to compare two values ​​with 0. So far, I'm just submitting your suggestions for optimizing the moreRelevant method.

Reply:
In Implementation Details of [DPTP-2743](https://issues.redhat.com/browse/DPTP-2743)
I have following bullet point:

- the the banch does not match, just prefer the "higher" based on string comparison
Now in my implementation it works by comparing 0 to 0